File: advanced/Conditional_Iterative/Sieve-of-Erathostain/sieve.py
import math
def sieve_of_eratosthenes(n):
    if n< 2:
        return "[empty]"
    lst= [x for x in range(2, n+1)]

    for i in reversed(lst):
        for j in range(2, int(i/2)+1):
            if i % j != 0:
                continue
            elif i % j == 0 :
                lst.remove(i)
                break
    

    return lst

print(sieve_of_eratosthenes(100))

File: advanced/Conditional_Iterative/Sieve-of-Erathostain/sieve.js

function sieveOfEratosthenes(n){
    let lst= []
    if (!n || n<2){return "[empty]"}
    for (let i=2; i<=n; i++){
        lst.push(i)
        }

    console.log(lst.filter(isPrime))
}

function isPrime(element) {
    for (let elementRange=2; element> elementRange; elementRange++){
        if (element % elementRange === 0){
            return false;
        }
    }
    return element;
};

sieveOfEratosthenes(100)

const demo = document.getElementById("demo")
demo.innerHTML= sieveOfEratosthenes(100)
